.github/workflows/lint.yaml

```diff
@@ -0,0 +1,18 @@
+name: lint
+on: [push, pull_request]
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install Python
```

Comment:
In a lot of other workflows, we cache the python environment to speed up action times https://github.com/marketplace/actions/cache. Would that be appropriate to incorporate here?

Reply:
IMO it's probably not worth it for this case - this should really only install 2 pkgs (updated pip + ruff), neither of which is particularly large thus is unlikely to be a bottleneck.